specs/trip_dispatch_spec.rb

```diff
@@ -89,4 +89,90 @@
       passenger.trips.must_include trip
     end
   end
+
+  describe "request_trip" do
+    before do
+
+      @trip_data = {
+        id: 601,
+        driver: RideShare::Driver.new(id: 3, name: "Lovelace", vin: "12345678912345678"),
+        passenger: RideShare::Passenger.new(id: 1, name: "Ada", phone: "412-432-7640"),
+        start_time: Time.now,
+        end_time: nil,
+        cost: nil,
+        rating: nil
+      }
+      @trip = RideShare::Trip.new(@trip_data)
+    end
+
+    it "Create a new instance of Trip" do
+      @trip.must_be_kind_of RideShare::Trip
+    end
+
+    it "Automatically assign the first driver whose status is :AVAILABLE to the trip Set the driver's status to :UNAVAILABLE" do
+      dispatcher = RideShare::TripDispatcher.new
+
+      first_driver_avalable_before = dispatcher.load_drivers[1]
+      first_driver_avalable_before.name.must_equal "Emory Rosenbaum"
+      first_driver_avalable_before.id.must_equal 2
+      first_driver_avalable_before.status.must_equal :AVAILABLE
+
+      first_driver_avalable = dispatcher.request_trip(1).driver
+      first_driver_avalable.name.must_equal "Emory Rosenbaum"
+      first_driver_avalable.id.must_equal 2
+      first_driver_avalable.status.must_equal :UNAVAILABLE
+
+    end
+
+    it "Update the end time, cost and rating to be nil" do
+      dispatcher = RideShare::TripDispatcher.new
```

Comment:
Since `dispatcher` appears in multiple places, it would be a good candidate for an instance variable in a `before` block. It will be recreated from scratch before every test, so you don't need to worry about keeping track of how many trips other tests request.